ui/src/components/TeamModal.vue

```diff
@@ -0,0 +1,271 @@
+      );
+      return data;
+    },
+    closeDialog() {
+      Object.assign(this.dialog, {
+        open: false,
+        title: '',
+        text: '',
+        action: '',
+      });
+    },
+    async getTeams(filters = this.filters) {
+      const response = await this.fetchTeams(this.page, this.itemsPerPage, filters);
+      if (response) {
+        this.pageCount = response.data.teams.pageInfo.numPages;
+        this.page = response.data.teams.pageInfo.page;
+        this.totalResults = response.data.teams.pageInfo.totalResults;
+      }
+      return response.data.teams.entities;
+    },
+    confirmDelete(item) {
+      Object.assign(this.dialog, {
+        open: true,
+        title: 'Delete this team?',
+        text: item.name,
+        action: () => this.delete(item),
+      });
+    },
+    findAndDelete(name, items = null, found = false) {
+      if (!found) {
+        if (!items) {
+          items = this.items;
+        }
+        items.forEach((child, index) => {
+          if (child.name == name) {
+            items.splice(index, 1);
+            found = true;
+          }
+          if (child.children) {
+            this.findAndDelete(name, child.children);
+          }
+        });
+      }
+    },
+    async delete(item) {
+      this.closeDialog();
+      try {
+        const response = await this.deleteTeam(item.name, this.organization);
+        if (response) {
+          this.findAndDelete(item.name);
+          this.$logger.debug(`Deleted team ${item.name}`);
+        }
+      } catch (error) {
+        Object.assign(this.dialog, {
+          open: true,
+          title: 'Error',
+          text: this.$getErrorMessage(error),
+          action: null,
+        });
+        this.$logger.error(`Error deleting team ${item.name}: ${error}`);
+      }
+    },
+    async add(parent = null) {
+      this.closeDialog();
+      try {
+        const response = await this.addTeam(this.teamName, this.organization, parent?.name);
+        if (response) {
+          if (parent) {
+            if (parent.children.length > 0) {
+              parent.children.push({ name: this.teamName, children: [] });
+            }
+          } else {
+            this.teams.push({ name: this.teamName, children: [] });
+          }
+          this.teamName = '';
+          this.$logger.debug(`Added team ${this.teamName}`);
+          return response;
+        }
+      } catch (error) {
+        Object.assign(this.dialog, {
+          open: true,
+          title: 'Error',
+          text: this.$getErrorMessage(error),
+          action: null,
+        });
+        this.$logger.error(`Error adding team ${this.teamName}: ${error}`);
+      }
+    },
+  },
+  async created() {
+    this.filters = { organization: this.organization };
+    const teamObjects = await this.getTeams();
+    teamObjects.forEach((team) => this.teams.push({ name: team.name, children: [] }));
```

Comment:
This looks a bit redundant, maybe adding one level of `subteams` to the query would make this initialization of `children` unnecessary and improve the UX. Right now, all teams have an arrow by their name, implying they have subteams, and when you click one that doesn't have children it looks like a bug. What do you think?

Reply:
It would help for just that one level, sure. But in the second level down there would still have to be the arrows because we dont know which of the children in the first level are leaf teams and which of them have subteams. The easiest way to make all of this simpler would be to just write a query for around five levels of subteams and construct the org tree on component creation. It would simplify a lot of the code in the rest of the functions as well. What do you think?

Reply:
You're right, it would only work for the first level, after that we'd have the same problem. What about using the `numchild` field for this? If it's other than 0, you can assume there are children and add the empty array then.
I agree that writing a query with multiple levels would simplify the code a lot, but we would have to set an artificial limit of levels somewhere, so I'm not sure about it.

Reply:
I believe we are working on a ticket to hide those fields as they fall under implementation details. Check -  https://github.com/chaoss/grimoirelab-sortinghat/issues/569, but having numChildren might be useful. We could check with Santiago about exposing just that field but we dont actually have it as part of the Teams model.

Reply:
I think new subteams are not being updated properly because of the condition on the `add` function on line 244 `if (parent.children.length > 0)`. It only adds the child if there are other children.

Reply:
otherwise if i dont have that condition , vue would  see that the children array is not empty so it would never call load children. none of the other subteams would show up

Reply:
an alternate is to call get subteams as part of the call for add team for the parent passed

Reply:
> an alternate is to call get subteams as part of the call for add team for the parent passed

This could work, can you try it?

Reply:
yep, sure can! And about using numchildren as a way of checking whether a team has any chlidren, where did we land on that? If you think its a good idea, I ll loop in Santiago to confirm from his side if its ok

Reply:
Yes, I think it's the best way to check for children if we don't want to limit the depth of subteams.

Reply:
Sure I ll put in the changes